packages/engine-core/src/data-proxy/DataProxyEngine.ts

```diff
@@ -4,28 +4,33 @@ import { Engine } from '../common/Engine'
 import type { EngineConfig, EngineEventType, GetConfigResult } from '../common/Engine'
 import { request } from './utils/request'
 import EventEmitter from 'events'
-import { createSchemaHash } from './utils/createSchemaHash'
 import { backOff } from './utils/backOff'
 import { getClientVersion } from './utils/getClientVersion'
+import { responseToError } from './errors/utils/responseToError'
+import { InvalidDatasourceError } from './errors/InvalidDatasourceError'
+import { NotImplementedYetError } from './errors/NotImplementedYetError'
+import { ForcedRetryError } from './errors/ForcedRetryError'
+import { SchemaMissingError } from './errors/SchemaMissingError'
+import { DataProxyError } from './errors/DataProxyError'
 // import type { InlineDatasources } from '../../../client/src/generation/utils/buildInlineDatasources'
 // TODO this is an issue that we cannot share types from the client to other packages
 
 const randomDebugId = Math.ceil(Math.random() * 1000)
 
-const MAX_RETRIES = 5
+const MAX_RETRIES = 10
```

Comment:
This increases the amount of retries, while the retry interval has been reduced (to `50ms`). That is for Vercel Edge Functions to retry as fast as possible since we have a limited interaction time of `1.5s`.